pkg/server/plugin/datastore/sql/migration.go

```diff
@@ -27,6 +40,23 @@ func migrateDB(db *gorm.DB, dbType string, log hclog.Logger) (err error) {
 		return initDB(db, dbType, log)
 	}
 
+	// TODO related epic https://github.com/spiffe/spire/issues/1083
+	// We do not allow breaking changes in 0.X beyond version 0.9.0, but
+	// 1.0+ may have such changes. Thus iff Major version is 0, we are
+	// compatible. This logic will change in 1.0+.
+	// With 1.0 code, we will need a proper version check between code and db versions.
```

Comment:
The stance here is that we promise no breaking changes within any given +/- 1 code version window. For example, given version 0.9, we promise that you will be backwards compatible to 0.8 and forwards compatible to 0.10... 0.11 may introduce changes that break 0.9, but not 0.10, and so forth.

Of course, the comparison we make to determine this compatibility is different between pre-1.0 and post-1.0 versions, and we are stopping short of solving for the latter. So, I recommend we update this comment, something like this
```
// TODO related epic https://github.com/spiffe/spire/issues/1083
// The version comparison logic in this package is specific to pre-1.0 versioning semantics.
// It will need to be updated prior to releasing 1.0. Ensure that we're still building a pre-1.0
// version before continuing, and fail cowardly if we're not.
```